src/main/java/com/google/growpod/controllers/GardenDao.java

```diff
@@ -118,4 +119,104 @@ public Garden getGardenById(String id) {
 
     return plantList;
   }
+
+  /**
+   * Adds a plant to a garden's plant list.
+   *
+   * @param gardenId the garden to add the plant to
+   * @param plant the plant object
+   * @return The plant's key
+   */
+  public String addPlant(String gardenId, Plant plant) {
+    // Add plant to plant list first
+    plant.setId("1"); // Dummy key to make .toEntity(datastoreInstance) work.
+    KeyFactory keyFactory = datastore.newKeyFactory().setKind("Plant");
+    IncompleteKey incompleteKey = keyFactory.newKey();
+
+    Key key = datastore.allocateId(incompleteKey);
+
+    Entity newEntity = Entity.newBuilder(key, plant.toEntity(datastoreInstance)).build();
+    datastore.add(newEntity);
+
+    // Then add relation
+    keyFactory = datastore.newKeyFactory().setKind("ContainsPlant");
+    incompleteKey = keyFactory.newKey();
+    key = datastore.allocateId(incompleteKey);
+
+    String plantId = Long.toString(newEntity.getKey().getId());
+    ContainsPlant relation = new ContainsPlant("1", gardenId, plantId);
+
+    newEntity = Entity.newBuilder(key, relation.toEntity(datastoreInstance)).build();
+    datastore.add(newEntity);
+    return plantId;
+  }
+
+  /**
+   * Deletes a user id from the garden's user list.
+   *
+   * @param gardenId the garden's id
+   * @param userId the user's id
+   * @return whether the query was successful.
+   */
+  public boolean deleteUser(String gardenId, String userId) {
+    // Existence check for key
+    Garden garden = getGardenById(gardenId);
+    if (garden == null) {
+      return false;
+    }
+
+    // Match user and garden
+    StructuredQuery<Entity> query =
+        Query.newEntityQueryBuilder()
+            .setKind("HasMember")
+            .setFilter(PropertyFilter.eq("garden-id", gardenId))
+            .build();
+    QueryResults<Entity> results = datastore.run(query);
+    if (!results.hasNext()) {
+      return false;
+    }
+    Entity entity = results.next();
+
+    datastore.delete(entity.getKey());
+    return true;
+  }
```

Comment:
Nice! I see it handles when there's only one user in the garden

Reply:
This solution itself actually has a bug in it. It gets fixed in #43, but I'll add it here as well